src/plonk/circuit.rs

```diff
@@ -479,6 +547,11 @@ impl<F: Field> ConstraintSystem<F> {
         index
     }
 
+    /// Query a selector at a relative position.
+    pub fn query_selector(&mut self, selector: Selector, at: Rotation) -> Expression<F> {
```

Comment:
If this querying logic was moved onto a gate-specific wrapper around `ConstraintSystem` (similar to how a `Region` will at some point wrap a `Layouter`), we could collect metadata about which columns / selectors are used in which gates (probably more easily than extracting that information from the gate expressions).